src/Analyzers/CSharp/CodeFixes/MakeMemberRequired/CSharpMakeMemberRequiredCodeFixProvider.cs

```diff
@@ -79,6 +79,10 @@ public override async Task RegisterCodeFixesAsync(CodeFixContext context)
             RegisterCodeFix(context, CSharpCodeFixesResources.Make_field_required, nameof(CSharpCodeFixesResources.Make_field_required));
         }
 
+        // The `required` modifier cannot be used if a member is not accessible from outside of type.
+        // For instance, having private required property in a public class leads to compiler error.
+        // This function checks whether the member can be accessed by checking containing type visibility (which is computed already taking into account whether the type is nested and what is its visibility based on that fact)
+        // against accessibility of member we are trying to make required
```

Comment:
by 'doc' i meant explaining why SymbolVisibility.Internal maps to Accessibility.Internal, bit SymbolVisibility.Private doesn't map to Private.  Thanks :)